spec/operators/switch-spec.js

```diff
@@ -47,10 +47,96 @@ describe('Observable.prototype.switch()', function () {
 
   it('should handle a hot observable of observables', function () {
     var x = cold(        '--a---b---c--|');
+    var xsubs =    '      ^       !';
     var y = cold(                '---d--e---f---|');
+    var ysubs =    '              ^             !';
     var e1 = hot(  '------x-------y------|', { x: x, y: y });
     var expected = '--------a---b----d--e---f---|';
     expectObservable(e1.switch()).toBe(expected);
+    expectSubscriptions(x.subscriptions).toBe(xsubs);
+    expectSubscriptions(y.subscriptions).toBe(ysubs);
+  });
+
+  it('should handle a hot observable of observables, outer is unsubscribed early', function () {
+    var x = cold(        '--a---b---c--|');
+    var xsubs =    '      ^       !';
+    var y = cold(                '---d--e---f---|');
+    var ysubs =    '              ^ !';
+    var e1 = hot(  '------x-------y------|', { x: x, y: y });
+    var unsub =    '                !';
+    var expected = '--------a---b---';
+    expectObservable(e1.switch(), unsub).toBe(expected);
+    expectSubscriptions(x.subscriptions).toBe(xsubs);
+    expectSubscriptions(y.subscriptions).toBe(ysubs);
+  });
+
+  it('should handle a hot observable of observables, inner never completes', function () {
+    var x = cold(        '--a---b---c--|');
+    var xsubs =    '      ^       !';
+    var y = cold(                '---d--e---f-----');
+    var ysubs =    '              ^               ';
+    var e1 = hot(  '------x-------y------|', { x: x, y: y });
+    var expected = '--------a---b----d--e---f-----';
+    expectObservable(e1.switch()).toBe(expected);
+    expectSubscriptions(x.subscriptions).toBe(xsubs);
+    expectSubscriptions(y.subscriptions).toBe(ysubs);
+  });
```

Comment:
@blesh I experimented using spaces instead of dashes for subscription marble diagrams. The dashes everywhere were making the whole test quite cluttered. Now the actual marble diagram for Observables are visually distinguishable from simple subscription marble diagrams. What do you think?